sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -1184,3 +1210,38 @@ func withUnmaskedPaths(unmaskPaths []string) oci.SpecOpts {
 		return nil
 	}
 }
+
+// Convert label to map[string]string for containerd.WithPullLabels.
+// Label are in the format of "key=value".
+func convertLabels(labels string) (map[string]string, error) {
+	labelsMap := make(map[string]string)
+	labelsList := strings.Split(labels, ",")
+	for _, label := range labelsList {
+		labelKeyValue := strings.Split(label, "=")
+
+		if len(labelKeyValue) != 2 {
+			return labelsMap, fmt.Errorf("invalid label format: %s. It should be key=value", label)
+		}
+		if err := Validate(labelKeyValue[0], labelKeyValue[1]); err != nil {
+			return labelsMap, err
+		}
+
+		labelsMap[labelKeyValue[0]] = labelKeyValue[1]
+	}
+	return labelsMap, nil
+}
+
+// Validate a label's key and value are under 4096 bytes
+//
+// validation logic derived from:
+// https://github.com/containerd/containerd/blob/f5e7fe0cb6286333a597a407029fe7c760626669/pkg/labels/validate.go#L32-L41
+func Validate(k, v string) error {
```

Comment:
The function should be named `validateLabel` or something to identify what it's validating.

I'd say just do the length check in `convertLabels` prior to the split on `=` and omit the `keyMaxLen` logic.

The main point from an interface standpoint is that the total length has to be under 4096 or it's not valid. We don't need to "derive" code from another project to check that.

Reply:
We don't need "derive". I already added this to `convertLable` function.